# Patch from https://aur.archlinux.org/cgit/aur.git/tree/0001-only-build-the-module-in-dkms.conf.patch?h=zfs-dkms
# Avoids recompiling ZFS userland utilities with DKMS rebuilds
From b4a2c0b184c9c9599421b15a430fb88deb5dbd17 Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz@archlinux.org>
Date: Sun, 28 Oct 2018 15:01:58 -0400
Subject: [PATCH] only build the module in dkms.conf

---
 scripts/dkms.mkconf | 19 ++-----------------
 1 file changed, 2 insertions(+), 17 deletions(-)

diff --git a/scripts/dkms.mkconf b/scripts/dkms.mkconf
index 88c289383..5a859a0e0 100755
--- a/scripts/dkms.mkconf
+++ b/scripts/dkms.mkconf
@@ -28,14 +28,7 @@ PACKAGE_CONFIG="${pkgcfg}"
 PRE_BUILD="configure
   --prefix=/usr
   --with-config=kernel
-  --with-linux=\$(
-    if [ -e "\${kernel_source_dir/%build/source}" ]
-    then
-      echo "\${kernel_source_dir/%build/source}"
-    else
-      echo "\${kernel_source_dir}"
-    fi
-  )
+  --with-linux="\${kernel_source_dir}"
   --with-linux-obj="\${kernel_source_dir}"
   \$(
     [[ -n \"\${ICP_ROOT}\" ]] && \\
@@ -69,7 +54,7 @@ POST_BUILD="scripts/dkms.postbuild
 "
 AUTOINSTALL="yes"
 REMAKE_INITRD="no"
-MAKE[0]="make"
+MAKE[0]="make -C module/"
 STRIP[0]="\$(
   [[ -r \${PACKAGE_CONFIG} ]] \\
   && source \${PACKAGE_CONFIG} \\
-- 
2.19.1

